module/os/linux/spl/spl-taskq.c

```diff
@@ -1041,23 +1050,38 @@ taskq_create(const char *name, int nthreads, pri_t pri,
 	if (flags & TASKQ_THREADS_CPU_PCT) {
 		ASSERT(nthreads <= 100);
 		ASSERT(nthreads >= 0);
-		nthreads = MIN(nthreads, 100);
-		nthreads = MAX(nthreads, 0);
-		nthreads = MAX((num_online_cpus() * nthreads) / 100, 1);
+		proper_nthreads = MIN(nthreads, 100);
+		proper_nthreads = MAX(proper_nthreads, 0);
+		proper_nthreads = MAX((num_online_cpus() * proper_nthreads) /
+		    100, 1);
 	}
 
 	tq = kmem_alloc(sizeof (*tq), KM_PUSHPAGE);
 	if (tq == NULL)
 		return (NULL);
 
+#ifdef HAVE_CPU_HOTPLUG
+	if (nthreads == boot_ncpus || flags & TASKQ_THREADS_CPU_PCT) {
```

Comment:
Having `nthreads == boot_ncpus` imply that we want to change the number of threads as cpu's come online is not especially intuitive.  Take for example the ZIO taskqs that have N threads.  If N happens to be boot_ncpus, you'll get more than N threads after hot-adding CPU's.

For cases where we do want to have "number of online CPU's" threads, I wonder if we could use TASKQ_THREADS_CPU_PCT with a value of 100%?  Or if that doesn't work then have an explicit flag to request NCPU's threads?

Reply:
We probably should use CPU_PCT with 100; I'll verify that works and change the callers where we pass boot_ncpus